pmd-core/src/main/java/net/sourceforge/pmd/RuleBuilder.java

```diff
@@ -0,0 +1,246 @@
+
+
+    public void rulesetName(String rulesetName) {
+        this.rulesetName = rulesetName;
+    }
+
+
+    public void message(String message) {
+        this.message = message;
+    }
+
+
+    public void defineProperty(PropertyDescriptor<?> descriptor) {
+        definedProperties.add(descriptor);
+    }
+
+
+    public void setDeprecated(boolean deprecated) {
+        isDeprecated = deprecated;
+    }
+
+
+    public void description(String description) {
+        this.description = description;
+    }
+
+
+    public void addExample(String example) {
+        examples.add(example);
+    }
+
+
+    public void priority(int priorityString) {
+        this.priority = RulePriority.valueOf(priorityString);
+    }
+
+
+    // Must be loaded after rule construction to know the Language
+    private void loadLanguageMinMaxVersions(Rule rule) {
+
+        if (minimumVersion != null) {
+            LanguageVersion minimumLanguageVersion = rule.getLanguage().getVersion(minimumVersion);
+            if (minimumLanguageVersion == null) {
+                throwUnknownLanguageVersionException("minimum", minimumVersion);
+            } else {
+                rule.setMinimumLanguageVersion(minimumLanguageVersion);
+            }
+        }
+
+        if (maximumVersion != null) {
+            LanguageVersion maximumLanguageVersion = rule.getLanguage().getVersion(maximumVersion);
+            if (maximumLanguageVersion == null) {
+                throwUnknownLanguageVersionException("maximum", maximumVersion);
+            } else {
+                rule.setMaximumLanguageVersion(maximumLanguageVersion);
+            }
+        }
+
+        checkLanguageVersionsAreOrdered(rule);
+    }
+
+
+    private void throwUnknownLanguageVersionException(String minOrMax, String unknownVersion) {
+        throw new IllegalArgumentException("Unknown " + minOrMax + " Language Version '" + unknownVersion
+                                           + "' for Language '" + language.getTerseName()
+                                           + "' for Rule " + name
+                                           + "; supported Language Versions are: "
+                                           + LanguageRegistry.commaSeparatedTerseNamesForLanguageVersion(language.getVersions()));
+    }
+
+
+    public Rule build() throws ClassNotFoundException, IllegalAccessException, InstantiationException {
+
+
+        Rule rule = (Rule) RuleBuilder.class.getClassLoader().loadClass(clazz).newInstance();
+
+        rule.setName(name);
+        rule.setRuleClass(clazz);
+
+        if (rule.getLanguage() == null) {
+            rule.setLanguage(language);
+        }
+
+        loadLanguageMinMaxVersions(rule);
+        rule.setSince(since);
+        rule.setMessage(message);
+        rule.setExternalInfoUrl(externalInfoUrl);
+        rule.setDeprecated(isDeprecated);
+        rule.setDescription(description);
+        rule.setPriority(priority == null ? RulePriority.valueOf(666) : priority);
```

Comment:
666?

Reply:
I don't remember why I did that ^^ Apparently I was trying to get RulePriority.LOW